backend/gce/api.go

```diff
@@ -142,7 +165,113 @@ func (api *gceAPI) insertRoute(subnet string) (*compute.Operation, error) {
 	return api.computeService.Routes.Insert(api.project, route).Do()
 }
 
-func (api *gceAPI) pollOperationStatus(operationName string) error {
+// refresh the held instance with the most recent information
+func (api *gceAPI) refreshInstance() error {
+	instance, err := api.computeService.Instances.Get(api.project, api.instanceZone, api.instanceName).Do()
+	if err != nil {
+		return err
+	}
+	api.gceInstance = instance
+	return nil
+}
+
+// combine ranges by name, updating any existing entries
+func combineSecondaryRanges(ranges []*compute.SubnetworkSecondaryRange, newRange *compute.SubnetworkSecondaryRange) []*compute.SubnetworkSecondaryRange {
+	m := make(map[string]*compute.SubnetworkSecondaryRange)
+
+	for i, secondaryRange := range ranges {
+		m[secondaryRange.RangeName] = ranges[i]
+	}
+
+	m[newRange.RangeName] = newRange
+
+	combined := make([]*compute.SubnetworkSecondaryRange, 0)
```

Comment:
nitpick: the slice can be initialized with the capacity needed.

```suggestion
	combined := make([]*compute.SubnetworkSecondaryRange, 0, len(m))
```